client/interface.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+
+	"github.com/google/trillian"
+)
+
+// LogVerifier is a client that verifies output from Trillian.
```

Comment:
LogVerifier doesn't seem like a good name as it implies a log reader whereas the API defines a writer. Also AddLeaf does a lot more work than just adding a leaf but I can't think of a better name right now.

I think we can solve the 'STR' naming by using 'Root'. The interface name includes 'Log' already.

```
type VerifyingLogClient interface {
  AddLeaf(ctx context.Context, data []byte) error
￼  UpdateRoot(ctx context.Context) error
￼  Root() trillian.SignedLogRoot
￼}
```

There's a lot of extra overhead compared to a simple client that doesn't get proofs so we should be clear what the indended use is. The documentation should include a description of AddLeaf and how it blocks and is not guaranteed to get a proof result within the specified deadline.